src/Lean/Server/Rpc/Deriving.lean

```diff
@@ -134,115 +110,93 @@ private def deriveStructureInstance (indVal : InductiveVal) (params : Array Expr
     -- helpers for type name syntax
     let paramIds ← params.mapM fun p => return mkIdent (← getFVarLocalDecl p).userName
     let typeId := Syntax.mkApp (← `(@$(mkIdent indVal.name))) paramIds
-    let packetId ← mkIdent <$> mkFreshUserName `RpcEncodingPacket
-    let packetAppliedId := Syntax.mkApp packetId uniqFieldEncIds
-
-    `(variable $binders*
-
-      structure $packetId:ident where
-        $[($fieldIds : $fieldEncIds)]*
-        deriving $(mkIdent ``FromJson), $(mkIdent ``ToJson)
-
-      instance : $(mkIdent ``RpcEncoding) $typeId $packetAppliedId where
-        rpcEncode a := return {
-          $[$encInits],*
-        }
-        rpcDecode a := return {
-          $[$decInits],*
-        }
+    let packetId := mkIdent <| indVal.name ++ `RpcEncodingPacket
```

Comment:
The name was intentionally hygienic -- it's only supposed to be obtainable using typeclass synthesis, i.e. through the `RpcEncoding` mapping. Do you need it to be accessible?

Reply:
I've found it very useful to be able to e.g. `#print TaggedText.RpcEncodingPacket` which is something that seems completely impossible with inaccessible names.  But I can revert this change if you prefer.

Reply:
That would still be a pretty awkward workflow, no? I usually use `trace.Elab.command` for debugging such macros fwiw.

Reply:
Yes, I'm using `trace.Elab.command` as well.  But it doesn't really help you to figure out what's going on with `variable (α β γ δ ε) [RE α α'] [RE β β'] ... in inductive Blah`.  (That is, what parameters the inductive actually gets.)  With definitions it's even worse because you only see the input syntax, but not the elaborated definition (which is simpler in inductives).